packages/create-amplify/src/package-manager-controller/yarn_classic_package_manager_controller.ts

```diff
@@ -0,0 +1,53 @@
+import { execa as _execa } from 'execa';
+import { executeWithDebugLogger } from '../execute_with_logger.js';
+import { type PackageManager } from '../package_manager.js';
+import {
+  DependencyType,
+  PackageManagerController,
+} from './package_manager_controller.js';
+
+/**
+ *
+ */
+export class YarnClassicPackageManagerController
+  implements PackageManagerController
+{
+  /**
+   * Abstraction around yarn classic commands that are needed to initialize a project and install dependencies
+   */
+  constructor(
+    private readonly projectRoot: string,
+    private readonly packageManager: PackageManager,
+    private readonly execa = _execa
+  ) {}
+
+  /**
+   * Installs the given package names as devDependencies
+   */
+  installDependencies = async (
+    packageNames: string[],
+    type: DependencyType
+  ): Promise<void> => {
+    const args = [`${this.packageManager.installCommand}`].concat(
+      ...packageNames
+    );
+    if (type === 'dev') {
+      args.push('-D');
+    }
+
+    try {
+      await executeWithDebugLogger(
+        this.projectRoot,
+        this.packageManager.executable,
+        args,
+        this.execa
+      );
+    } catch {
+      throw new Error(
+        `\`${this.packageManager.executable} ${args.join(
+          ' '
+        )}\` did not exit successfully.`
+      );
```

Comment:
This try/catch can be removed as it doesn't add any additional information that isn't already in the error message from `executeWithDebugLogger` (also applies to other package manager controllers)

Reply:
Done b5665ec7d